views/partials/asset.handlebars

```diff
@@ -1,29 +1,37 @@
-{{#ifConditional fileInfo.fileType '===' 'video/mp4'}}
-    {{#ifConditional fileInfo.fileExt '===' 'gifv'}}
-        <video class="gifv-show" autoplay loop muted>
-            <source src="/{{fileInfo.claimId}}/{{fileInfo.name}}.{{fileInfo.fileExt}}">
-            {{!--fallback--}}
-            Your browser does not support the <code>video</code> element.
-        </video>
-    {{else}}
+<div id="asset-display-component">
+    <div id="asset-status">
+        <div id="searching-message" hidden="true">
+            <p>Sit tight, we're searching the LBRY blockchain for your asset!</p>
+            {{> progressBar}}
+            <p>Curious what magic is happening here? <a class="link--primary" target="blank" href="https://lbry.io/faq/what-is-lbry">Learn more.</a></p>
+        </div>
+        <div id="failure-message" hidden="true">
+            <p>Unfortunately, we couldn't download your asset from LBRY.  You can help us out by sharing the below error message in the <a class="link--primary" href="https://discord.gg/YjYbwhS" target="_blank">LBRY discord</a>.</p>
+            <i><p id="error-message"></p></i>
+        </div>
+    </div>
+    <div id="asset-holder" hidden="true">
+        {{#ifConditional claimInfo.contentType '===' 'video/mp4'}}
+            {{> video}}
+        {{else}}
+            {{> image}}
+        {{/ifConditional}}
+        <div>
+            <a id="asset-boilerpate" class="link--primary fine-print" href="/{{claimInfo.claimId}}/{{claimInfo.name}}">hosted via Spee&lt;h</a>
+        </div>
 
-            <video id="video-player" class="video-show video" controls poster="{{fileInfo.thumbnail}}">
-                <source src="/{{fileInfo.claimId}}/{{fileInfo.name}}.{{fileInfo.fileExt}}">
-                {{!--fallback--}}
-                Your browser does not support the <code>video</code> element.
-            </video>
-            <script type="text/javascript">
-                document.addEventListener('DOMContentLoaded', resizeVideoPlayer)
-                window.addEventListener("resize", resizeVideoPlayer);
-                function resizeVideoPlayer() {
-                    const div = document.getElementById('video-player');
-                    const width = div.offsetWidth;
-                    div.height = (width * 9 / 16);
-                }
-            </script>
-    {{/ifConditional}}
-{{else}}
-    <a href="/{{fileInfo.claimId}}/{{fileInfo.name}}.{{fileInfo.fileExt}}">
-        <img  class="image-show" src="/{{fileInfo.claimId}}/{{fileInfo.name}}.{{fileInfo.fileExt}}" />
-    </a>
-{{/ifConditional}}
\ No newline at end of file
+    </div>
+</div>
+
+<script type="text/javascript">
+
+    const asset = new Asset();
```

Comment:
Among the questions we'll discuss: why not just...
```
const asset = new Asset({
  src: "foo",
  claimId: "bar",
  ...
})
```

Reply:
Also, given that asset is used elsewhere in code, we might want this class to have a more specific name.